engine/schema/src/com/cloud/dc/dao/DataCenterVnetDaoImpl.java

```diff
@@ -100,29 +101,42 @@ public void lockRange(long dcId, long physicalNetworkId, Integer start, Integer
     }
 
     @Override
-    public List<DataCenterVnetVO> findVnet(long dcId, String vnet) {
-        SearchCriteria<DataCenterVnetVO> sc = VnetDcSearch.create();
-        ;
-        sc.setParameters("dc", dcId);
-        sc.setParameters("vnet", vnet);
-        return listBy(sc);
-    }
-
-    @Override
     public int countZoneVlans(long dcId, boolean onlyCountAllocated) {
         SearchCriteria<Integer> sc = onlyCountAllocated ? countAllocatedZoneVlans.create() : countZoneVlans.create();
         sc.setParameters("dc", dcId);
         return customSearch(sc, null).get(0);
     }
 
+    private List<DataCenterVnetVO> findOverlappingVnets(final long dcId, final Long physicalNetworkId, final String vnet) {
+        final List<Integer> searchVnets = UriUtils.expandVlanUri(vnet);
+        final List<DataCenterVnetVO> overlappingVnets = new ArrayList<>();
+        if (searchVnets != null && searchVnets.size() > 0) {
```

Comment:
what about inverting this conditional?
`if( searchVnets == null || searchVnets.size() == 0){return overlappingVnets;}`
This helps to reduce the number of IFs inside of IFs

Reply:
Fixed, thanks.